File: ljh/level2/짝지어 제거하기/DeletePair.java
package Programmers.test;

import java.util.Stack;

/**
 * 짝지어 제거하기 (12973)
 */

class DeletePair {
    public int solution(String s) {
        char[] arr = s.toCharArray();
        Stack<Character> stack = new Stack<>();

        for (int i=0; i<arr.length; i++) {
            if(stack.isEmpty()) {
                stack.push(arr[i]);
                continue;
            }
            if (stack.peek() == arr[i]) {
                stack.pop();
            } else {
                stack.push(arr[i]);
            }
        }

        return stack.isEmpty()? 1 : 0;
    }
}

File: ljh/level2/짝지어 제거하기/DeletePair.js
// 짝지어 제거하기 (12973)
function solution(s) {
    let stack = [];
    for (let e of s) {
        let length = stack.length;
        if (length === 0) {
            stack.push(e);
            continue;
        }
        if (stack[stack.length-1] === e) {
            stack.pop();
        } else {
            stack.push(e);
        }
    }

    return stack.length === 0? 1 : 0;
}